src/solvers/prop/prop_conv_solver.cpp

```diff
@@ -493,3 +493,51 @@ std::size_t prop_conv_solvert::get_number_of_solver_calls() const
 {
   return prop.get_number_of_solver_calls();
 }
+
+const char *prop_conv_solvert::context_prefix = "prop_conv::context$";
+
+void prop_conv_solvert::set_to(const exprt &expr, bool value)
+{
+  if(context_literals.empty())
+  {
+    // We are in the root context.
+    do_set_to(expr, value);
+  }
+  else
+  {
+    // We have a child context. We add context_literal ==> expr to the formula.
+    do_set_to(or_exprt(literal_exprt(!context_literals.back()), expr), value);
```

Comment:
So here we set the whole `or_expr` to either true or false. If `value=true` then we add `~context_literal \/ expr` but if `value=false` we add `context_literal /\ ~expr`. Or I am reading this wrongly?